__tests__/app.test.js

```diff
@@ -244,7 +244,202 @@ describe.only('POST /api/articles/:article_id/comments', () => {
+            })
+         })
+    })
+})
+
+describe('/api/articles - featured sorts articles by any valid column in a specified order', () => {
+    test('GET-200, a query sorted by title and order by acsending', () => {
+        return request(app)
+        .get('/api/articles?sort_by=title&order=asc')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles).toBeSortedBy('title',{
+                ascending: true })
+            });
+        });
+    test('GET-200, a query sorted by comment count and order by descending', () => {
+        return request(app)
+        .get('/api/articles?sort_by=comment_count&order=desc')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles).toBeSortedBy('comment_count',{descending: true, coerce: true, })
+                });
+            });
+    test('GET-400, responds with an error for invalid column sort by query', () => {
+        return request(app)
+        .get('/api/articles?sort_by=invalid_column')
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe('bad request');
+            });
+        });
+    test('GET-400, responds with an error for invalid order query', () => {
+        return request(app)
+        .get('/api/articles?sort_by=created_at&order=invalid_order')
+        .expect(400)
+        .then(({ body }) => {
+             expect(body.msg).toBe('bad request');
+                });
+        });
     })
 
+describe('/api/articles - topic query', () => {
+    test('GET - 200 takes a topic query and responds with all articles which are filtered out by a specified value - cats', () => {
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles.length).toBe(1);
+            body.articles.forEach((article) => {
+                expect(article.topic).toBe('cats');         
+            })
+        })
+    });
+    test('GET - 200 responds with all articles which are filtered out by a specified value - mitch', () => {
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles.length).toBe(12);
+            body.articles.forEach((article) => {
+                expect(article.topic).toBe('mitch');         
+            })
+        })
+    });
+    test('GET - 400, responds with an error when topic has empty value', () => {
+        return request(app)
+        .get('/api/articles?topic=')
+        .expect(400)
+        .then(({ body }) => {
+             expect(body.msg).toBe('bad request');
+        });
+    }); 
+    test('GET - 404, responds with an error when topic value does not exist', () => {
+        return request(app)
+        .get('/api/articles?topic=dogs')
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe('not found');
+        });
+    });             
```

Comment:
Really good job considering all the tests in this suite!

One untested behaviour here:

- Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper.